framework/encode/vulkan_state_tracker.cpp

```diff
@@ -663,30 +663,6 @@ void VulkanStateTracker::TrackUpdateDescriptorSets(uint32_t                    w
             {
                 auto& binding = wrapper->bindings[current_binding];
 
-                binding.write_pnext = nullptr;
-                binding.write_pnext_memory.Reset();
-                if (write->pNext != nullptr)
-                {
-                    binding.write_pnext = TrackStruct(write->pNext, &binding.write_pnext_memory);
-                    auto* pnext         = reinterpret_cast<const VkBaseInStructure*>(binding.write_pnext);
-                    switch (pnext->sType)
-                    {
-                        case VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET_ACCELERATION_STRUCTURE_KHR:
-                        {
-                            auto* acc_struct = reinterpret_cast<VkWriteDescriptorSetAccelerationStructureKHR*>(
-                                const_cast<void*>(binding.write_pnext));
-                            binding.record_write_set_accel_structs.clear();
-                            std::copy(acc_struct->pAccelerationStructures,
-                                      acc_struct->pAccelerationStructures + acc_struct->accelerationStructureCount,
-                                      std::back_inserter(binding.record_write_set_accel_structs));
-                            acc_struct->pAccelerationStructures = binding.record_write_set_accel_structs.data();
```

Comment:
`binding.record_write_set_accel_structs` and `binding->acceleration_structures` are duplicated. It's good to remove `binding.record_write_set_accel_structs` and use `binding->acceleration_structures`. 

But my concern is to remove `binding.write_pnext = TrackStruct(write->pNext, &binding.write_pnext_memory);` and `write.pNext = binding->write_pnext;`. The original code is to copy and write all pnexts. But this PR become to only write `VkWriteDescriptorSetAccelerationStructureKHR`. The others are ignored. I'm not sure if it's safety. Should we keep that copying and writing all pnexts? or only `VkWriteDescriptorSetAccelerationStructureKHR`?

Reply:
@locke-lunarg

that's a good point, my reasoning to remove all raw pNext-data was this:

if not `nullptr`, we can expect `pNext` to either hold:
- a `VkWriteDescriptorSetAccelerationStructureKHR` (and we're already saving that data in `binding->acceleration_structures` - so all good)
- a `VkWriteDescriptorSetAccelerationStructureNV` (we do not support it and it's unclear/debatable if we should)
- a `VkWriteDescriptorSetInlineUniformBlock` -> this is addressed in a follow-up [PR](https://github.com/LunarG/gfxreconstruct/pull/1504), adding support for it.

I hence argue we got this covered, both relevant cases can do without the raw pNext data,
the follow-up PR should clear things up and cover the missing inline-uniform-block case.